moveit2/run.sh

```diff
@@ -12,5 +12,22 @@ IMG_NAME=openrobotics/moveit2
 CONTAINER_NAME="$(tr '/' '_' <<< "$IMG_NAME")"
 
 # Start the container
-docker run --rm -it --name $CONTAINER_NAME  --network host \
-    -e DISPLAY -e TERM   -e QT_X11_NO_MITSHM=1 $IMG_NAME
+# --rm: delete container after exiting
+# -it: Interactive TTY mode so you can use a shell.
+# --network host: Does not isolate Docker container from host's network interfaces.
+#                 Allows all traffic to pass through as if it originated from the host.
+#                 In our case, useful for DDS traffic for ROS.
+# -e DISPLAY: Pass the X11 display through to the container.
+# -e TERM: Pass the kind of terminal being used through to the container.
+# -e QT_X11_NO_MITSHM=1: Disables shared memory extension for Qt/X11, which does not work consistently inside a process namespace like a container.
+# -v "$XAUTHORITY:/.Xauthority": the $XAUTHORITY environment variable on the host contains the path to a file that grants access to X11.
+#                                We mount the path defined in that environment variable into the container at /.Xauthority.
+# -e XAUTHORITY=/.Xauthority: We set $XAUTHORITY *inside* the container to the place we just mounted the Xauthority (/.Xauthority)
+docker run --rm -it --name $CONTAINER_NAME \
+                    --network host \
+                    -e DISPLAY \
+                    -e TERM \
+                    -e QT_X11_NO_MITSHM=1 \
+                    -v "$XAUTHORITY:/.Xauthority" \
```

Comment:
Oh goodness. We've entered a world of pain here with user permissions, partially because the `space-ros` user has UID/GID 1001 (because of the ubuntu user in noetic?). My host's user is 1000, so this wonderful thing happens:

```
spaceros-user@ubuntu-linux-22-04-02-desktop:~/moveit2$ id
uid=1001(spaceros-user) gid=1001(spaceros-user) groups=1001(spaceros-user),27(sudo)
spaceros-user@ubuntu-linux-22-04-02-desktop:~/moveit2 $ ls -al /.Xauthority 
-rw------- 1 ubuntu ubuntu 146 May 28 14:14 /.Xauthority
```

We could of course map things with `-u "$(id -u):$(id -g)"` but of course it screws up built file permissions.

🫠

Reply:
Should we just go with the trusty nuclear option of modifying xhost at start and at exit?

Reply:
lets do this https://github.com/space-ros/space-ros/pull/264 and kick the problem down the road